modules/sfp_c99.py

```diff
@@ -0,0 +1,624 @@
+                or region is not None
+                or city is not None
+                or postalCode is not None
+            ):
+                evt = SpiderFootEvent(
+                    "GEOINFO",
+                    f"Continent: {continent}, Country: {country}, Region: {region}, City: {city}, Postal code: {postalCode}",
+                    self.__name__,
+                    event,
+                )
+                self.notifyListeners(evt)
+
+    def emitSkypeResolverData(self, data, event):
+        self.emitRawRirData(data, event)
+
+        IP = data.get("ip")
+        IPs = data.get("ips")
+
+        if IP is not None and IP not in IPs:
+            evt = SpiderFootEvent(
+                "IP_ADDRESS",
+                IP,
+                self.__name__,
+                event,
+            )
+            self.notifyListeners(evt)
+
+        if isinstance(IPs, list):
+            for IPElem in IPs:
+                if self.checkForStop():
+                    return None
+
+                evt = SpiderFootEvent(
+                    "IP_ADDRESS",
+                    IPElem.strip(),
+                    self.__name__,
+                    event,
+                )
+                self.notifyListeners(evt)
+
+    def emitWafDetectorData(self, data, event):
+        self.emitRawRirData(data, event)
+        firewall = data.get("result")
+
+        if firewall is not None:
+            evt = SpiderFootEvent(
+                "WEBSERVER_TECHNOLOGY",
+                f"Firewall detected: {firewall}",
+                self.__name__,
+                event,
+            )
+            self.notifyListeners(evt)
+
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        # Once we are in this state, return immediately.
+        if self.errorState:
+            return None
+
+        self.sf.debug(f"Received event, {eventName}, from {srcModuleName}")
+
+        if self.opts["api_key"] == "":
+            self.sf.error("You enabled sfp_c99, but did not set an API key!", False)
+            self.errorState = True
+            return None
+
+        if (
+            self.opts["response_type"].lower() != "plain"
+            and self.opts["response_type"].lower() != "json"
+        ):
+            self.sf.info(
+                "You enabled sfp_c99 ,but did not set response type to json or plain! JSON will be used by default"
+            )
+            self.opts["response_type"] = "json"
+
+        # Don't look up stuff twice
+        if eventData in self.results:
+            self.sf.debug(f"Skipping {eventData}, already checked.")
+            return None
+
+        self.results[eventData] = True
+
+        if eventName == "PHONE_NUMBER":
+            phoneData = self.query("phonelookup", "number", eventData)
+            phoneData = phoneData.get("details") if phoneData is not None else None
+
+            if phoneData is not None:
```

Comment:
I think it's safer to use `if phoneData` vs. `if phoneData is not None` for two reasons: 1) Empty strings, booleans and empty lists will not be `None` and evaluate as `True` here, so that might have unexpected consequences. 2) IMHO it's easier to read.

Reply:
Definitely agree 👍